### ISSUE TITLE
unrealizedPnl is always 0.0 for FTX positions, realizedPnL shows historical total

### ISSUE THREAD

**Issue by idclevxx** [ccxt/ccxt#7969]
unrealizedPnl is unattainable (always 0.0) via `private_get_account` and `private_get_positions` while realizedPnl shows historical total for each contract (not the position currently open). Cannot find another implicit call that gets unrealizedPnl.

Thanks for a great product
/Clev

**Comment by kroitor** [comment]
I'd suggest to report this to the FTX tech support. The more reports – the sooner they fix it in the API on their side.

**Comment by idclevxx** [comment]
Thanks @kroitor, will do.

**Comment by Rishirandhawa** [comment]
@idclevxx were you able to get any response from FTX ? Not sure if you faced any issue with entryPrice - it seems to be some price close to mark price and changes with every call even though I didn't add or subtract from the position.

**Comment by idclevxx** [comment]
@Rishirandhawa it crossed my mind whether it was intentional, I will look into it and report back.

**Comment by kroitor** [comment]
@idclevxx appreciate your feedback, thank you!

**Comment by tetouani63** [comment]
I found a link that explains the behavior [on ftx website](https://help.ftx.com/hc/en-us/articles/360027668732-Profit-Loss-Calculation).

Indeed it is intentional.

**Comment by wan930930** [comment]
Hi，everyone,I found a way to calculate the pnl in FTX.

As I have see on the FTX web platform, they have shown the pnl, breakevenprice, etc.
I tried to use
            params = {"showAvgPrice": True}
            coin_positions = exchange.privateGetPositions(params)
and finally made the position result showed more than simply exchange.privateGetPositions().

The extra informations are:
   'recentAverageOpenPrice': None,
   'recentPnl': None,
   'recentBreakEvenPrice': None,
   'cumulativeBuySize': None,
   'cumulativeSellSize': None

I suppose this 'recentPnl' term shows the pnl of most recent opening position.

Hope this would help. Good luck guys!

**Comment by frosty00** [comment]
> `Hi，everyone,I` found a way to calculate the pnl in FTX.
> 
> As I have see on the FTX web platform, they have shown the pnl, breakevenprice, etc. I tried to use
> 
> ```
>   params = {"showAvgPrice": True}
>   coin_positions = exchange.privateGetPositions(params)
> ```
> 
> and finally made the position result showed more than simply exchange.privateGetPositions().
> 
> The extra informations are:
> 
> ```
>    'recentAverageOpenPrice': None,
>    'recentPnl': None,
>    'recentBreakEvenPrice': None,
>    'cumulativeBuySize': None,
>    'cumulativeSellSize': None
> ```
> 
> I suppose this 'recentPnl' term shows the pnl of most recent opening position.
> 
> Hope this would help. Good luck guys!

yep it's in the latest version of ftx inside of `fetchPositions`
